api/src/main/java/org/pmiops/workbench/institution/InstitutionEmailAddressMapper.java

```diff
@@ -0,0 +1,35 @@
+package org.pmiops.workbench.institution;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.mapstruct.Mapper;
+import org.pmiops.workbench.db.model.DbInstitution;
+import org.pmiops.workbench.db.model.DbInstitutionEmailAddress;
+import org.pmiops.workbench.model.Institution;
+import org.pmiops.workbench.utils.mappers.MapStructConfig;
+
+@Mapper(config = MapStructConfig.class)
+public interface InstitutionEmailAddressMapper {
+  default Set<DbInstitutionEmailAddress> modelToDb(
+      final Institution modelInstitution, final DbInstitution dbInstitution) {
+    return Optional.ofNullable(modelInstitution.getEmailAddresses()).orElse(Collections.emptyList())
+        .stream()
+        .distinct()
+        .map(
+            address ->
+                new DbInstitutionEmailAddress()
+                    .setEmailAddress(address)
+                    .setInstitution(dbInstitution))
+        .collect(Collectors.toSet());
+  }
+
+  default List<String> dbToModel(Set<DbInstitutionEmailAddress> dbAddresses) {
```

Comment:
You should be able to just define a mapping from `DbInstitutionEmailAddress` to `String` and let MapStruct generate this for you. It's pretty smart with collection mappings.

Reply:
In this case I want the list to be sorted so I'd have to add a step for that anyway

Reply:
update: I was able to do this.  thanks.